client/app/domain/general-informations/domain-general-informations.controller.js

```diff
@@ -310,6 +314,33 @@ export default class DomainTabGeneralInformationsCtrl {
       });
   }
 
+  getAllOptionDetails(serviceName) {
+    this.loading.options = true;
+    return this.Domain.getOptions(serviceName)
+      .then(options => this.$q.all(
+        _.map(options, option => this.Domain.getOption(serviceName, option)
+          .then((optionDetail) => {
+            _.set(optionDetail, 'optionActivated', optionDetail.state === this.DOMAIN.DOMAIN_OPTION_STATUS.ACTIVE);
+            return optionDetail;
+          })),
+      ))
+      .then((options) => {
+        const transformedOptions = {};
+        _.forEach(options, (option) => {
+          transformedOptions[option.option] = option;
+        });
```

Comment:
could a `_.groupBy` suit the need here ? https://lodash.com/docs/4.17.11#groupBy

Reply:
https://lodash.com/docs/3.10.1#groupBy ;)

Reply:
I was actually looking for `_.keyBy` from https://lodash.com/docs/4.17.14#keyBy
Since this is not available in the version of lodash we are using, I thought I would do that manually :)

`_.groupBy` groups the elements of an array, where, each value in the returned object is an array. I just wanted each value in the returned object to be a single element from the array